packages/flutter_ai_providers/lib/src/providers/openwebui.dart

```diff
@@ -0,0 +1,307 @@
+    'image_url': imageUrl,
+    'name': name,
+  };
+
+  factory _OwuiImageAttachment.fromImageAttachment(ImageFileAttachment attachment) {
+    final base64Image = base64Encode(attachment.bytes);
+    return _OwuiImageAttachment(
+      type: 'image_url',
+      imageUrl: {'url': "data:${attachment.mimeType};base64,$base64Image"},
+      name: attachment.name,
+    );
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+class _OwuiFileAttachment {
+  final String type;
+  final String id;
+
+  _OwuiFileAttachment({
+    required this.type,
+    required this.id,
+  });
+
+  Map<String, dynamic> toJson() => {
+    'type': type,
+    'id': id,
+  };
+}
+
+/// A provider for [open-webui](https://openwebui.com/)
+/// Use open-webui as unified chat provider.
+class OpenwebuiProvider extends LlmProvider with ChangeNotifier {
+  /// Creates an [OpenwebuiProvider] instance with an optional chat history.
+  ///
+  /// The [history] parameter is an optional iterable of [ChatMessage] objects
+  /// representing the chat history
+  /// The [model] parameter is the ai model to be used for the chat.
+  /// The [baseUrl] parameter is the host of the open-webui server.
+  /// For example port 3000 on localhost use 'http://localhost:3000'
+  /// The [apiKey] parameter is the API key for the open-webui server.
+  /// See the [docs](https://docs.openwebui.com/) for more information.
+  /// Example:
+  /// ``` dart
+  /// LlmChatView(
+  ///   provider: OpenwebuiProvider(
+  ///     host: 'http://127.0.0.1:3000',
+  ///     model: 'llama3.1:latest',
+  ///     apiKey: "YOUR_API_KEY",
+  ///     history: [],
+  ///   ),
+  /// )
+  /// ```
+  OpenwebuiProvider({
+    Iterable<ChatMessage>? history,
+    required String model,
+    required String baseUrl,
+    String? apiKey,
+  }): _history = history?.toList() ?? [],
+      _model = model,
+      _host = baseUrl,
+      _apiKey = apiKey;
+
+  final List<ChatMessage> _history;
+  final String _model;
+  final String _host;
+  final String? _apiKey;
+  final List<_OwuiFileAttachment> _fileAttachments = [];
+  final List<_OwuiImageAttachment> _imageAttachments = [];
+  final _emptyMessage = ChatMessage(origin: MessageOrigin.llm, text: null, attachments: []);
+
+  @override
+  Stream<String> generateStream(
+    String prompt, {
+    Iterable<Attachment> attachments = const [],
+  }) async* {
+    final userMessage = ChatMessage(text: prompt, attachments: attachments, origin: MessageOrigin.user);
+    final llmMessage = ChatMessage(text: "", attachments: [], origin: MessageOrigin.llm);
+    
+    yield* _generateStream([userMessage, llmMessage]);
+    
+    notifyListeners();
+  }
+
+  @override
+  Stream<String> sendMessageStream(
+    String prompt, {
+    Iterable<Attachment> attachments = const [],
+  }) async* {
+    final userMessage = ChatMessage(text: prompt, attachments: attachments, origin: MessageOrigin.user);
+    final llmMessage = ChatMessage(text: null, attachments: [], origin: MessageOrigin.llm);
```

Comment:
You can use the `ChatMessage.user()` and `ChatMessage.llm()` factories.

Reply:
Done